django_x509/templates/admin/django_x509/change_form.html

```diff
@@ -22,4 +22,56 @@
     </div>
 </fieldset>
 {% endif %}
+
+<script>
+  (function($){
+    // function for operation_type switcher
+    var filter_div = function(operation_type){
+      //for each div in the form
+      $(".form-row").each(function(i, obj){
+        $(this).show()
+        obj_class = $(this).attr("class")
+        //if default
+        if (operation_type == ''){
+          if( obj_class != "form-row field-operation_type" && obj_class != "form-row field-name"){
+            $(this).hide()
+          }
+          }
+        //if import
+        if (operation_type == 'import'){
+          if(obj_class != "form-row field-certificate"
+          && obj_class != "form-row field-private_key"
+          && obj_class != "form-row field-operation_type"){
+            $(this).hide()
+          }
+        }
+        //if new
+        if (operation_type == 'new'){
+          if(obj_class == "form-row field-certificate"
+          || obj_class == "form-row field-private_key"
+          && obj_class != "form-row field-operation_type"){
+            $(this).hide()
+          }
+        }
+      })
+    }
+    //filter default
+    filter_div('')
+    $(".field-operation_type").on('change',function(){
+      switcher = $(".field-operation_type").find(":selected").text()
+      //filter if default
+      if(switcher==''){
+        filter_div('')
+      }
+      //filter if new
+      if(switcher=='new'){
+        filter_div('new')
+      }
+      //filter if import
+      if(switcher=='import'){
+        filter_div('import')
+      }
+    })
+  })(django.jQuery)
```

Comment:
This needs to be simplified, seek inspiration from this:
https://github.com/openwisp/django-loci/blob/master/django_loci/static/django-loci/js/floorplan-inlines.js

And:
https://github.com/openwisp/django-netjsonconfig/blob/master/django_netjsonconfig/static/django-netjsonconfig/js/switcher.js